### ISSUE TITLE
Allow to use window functions inside CASE statement.

### ISSUE THREAD

**Issue by viktor-klestov** [ClickHouse/ClickHouse#38911]
**How to reproduce**
```
SELECT CASE
           WHEN sum(col) over () > 0 THEN col + 1
           ELSE 0 END
FROM table
```

**Error message and/or stacktrace**

Code: 10. DB::Exception: Not found column plus(col, 1) in block. (NOT_FOUND_COLUMN_IN_BLOCK) (version 22.7.1.1164 (official build))

**Comment by alexey-milovidov** [comment]
@novikd please check why it's not fixed by #37848.

**Comment by novikd** [comment]
@alexey-milovidov it was already fixed in #39112. I checked that the following query works in master:
```sql
SELECT CASE
    WHEN sum(number) over () > 0 THEN number + 1
    ELSE 0 END
FROM numbers(10)
```

**Comment by alexey-milovidov** [comment]
@novikd Ok. I checked, and #39112 does not contain a test with CASE. Let's add it just in case.
